### PR TITLE
[ZEPPELIN-2865] upgrade Beam interpreter to latest version

### PR DESCRIPTION
**Pull request by XuMingmin**
### What is this PR for?
upgrade Beam interpreter to use the latest version of Apache Beam.

### What type of PR is it?
[Improvement]

### Todos
* 

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-2865

### How should this be tested?
* Start the Zeppelin server
* The prefix of interpreter is %beam and then write your code with required imports and the runner

Refer to `docs/interpreter/beam.md` for an example;

### Questions:
* Does the licenses files need update?  no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes, updated `docs/interpreter/beam.md` and `README.md`

### CODE REVIEW

**Comment by 1ambda** [comment]
Thanks for the contribution. Could you check the failed job?

- https://travis-ci.org/XuMingmin/zeppelin/jobs/264931044

**Comment by XuMingmin** [comment]
sure, seems an error in UI test, let me trigger it again.

File: beam/README.md [hunk 1]
```diff
@@ -8,12 +8,9 @@ Current interpreter implementation supports the static repl. It compiles the cod
 You have to first build the Beam interpreter by enable the **beam** profile as follows:
 
 ```
-mvn clean package -Pbeam -DskipTests
+mvn clean package -Pbeam -DskipTests -Pscala-2.10
```
Comment (1ambda) [inline beam/README.md]: Hi, you removed the `### Notice` section and added `-Pscala-2.10`. 
It's quite unclear to use scala 2.10 only or not. Does scala-2.10 is not mandatory as of beam 2.0?

Comment (XuMingmin) [inline beam/README.md]: It's still `scala-2.10` only in Beam 2.0, let me restore the notice message. --Btw, does Zeppelin provide a release version with `scala-2.11`(donot find in download page)? Wanna know what will happen with Beam.

Comment (1ambda) [inline beam/README.md]: @XuMingmin Hi Zeppelin can be built with scala-2.11

- https://github.com/apache/zeppelin/blob/master/dev/change_scala_version.sh

You can use it like

```
./dev/change_scala_version.sh 2.11
mvn clean package...
`

Comment (XuMingmin) [inline beam/README.md]: Thanks @1ambda . I verified the libraries in Beam, that it only supports scala_2.10 so far.

**Comment by 1ambda** [comment]
LGTM. Merge if no more discussion.

**Pull request closed without merging**